reqs/docs.pip

```diff
@@ -20,6 +20,10 @@ six
 snowballstemmer
 sphinx-rtd-theme
 sphinxcontrib-websupport
+sphinx-gallery
+sphinx-code-tabs
 sympy
 typing
 urllib3
+torch>=1.13.0
+pillow
```

Comment:
Because we are not running the `magnitude_pruning.py` file while generating; that is done in the internal CI. 

We need `torch` because the sphinx auto-gen code does import `coremltools/optimize/torch/__init__.py` which does a couple imports including using `torch`.

For some reason, `sphinx-gallery` generates some images and uses `pillow`.

Reply:
Okay, I thought they will require those decencies in the tutorial python file in order to generate the file.
Good to know. Thanks.